datafusion/datasource-parquet/src/opener.rs

```diff
@@ -121,23 +121,40 @@ impl FileOpener for ParquetOpener {
         let enable_page_index = self.enable_page_index;
 
         Ok(Box::pin(async move {
-            // Don't load the page index yet - we will decide later if we need it
-            let options = ArrowReaderOptions::new().with_page_index(false);
-
+            // Don't load the page index yet. Since it is not stored inline in
```

Comment:
this is all comments and renaming variables so they better reflect what they are (`metadata` --> `reader_metadata` for example to distinguish between `ParquetMetaData`)